websites/K/Ko-Fi/presence.ts

```diff
@@ -0,0 +1,172 @@
+      .startsWith("/account/externallogincallback")
+  )
+    presenceData.details = "Logining in...";
+  else if (document.location.pathname.toLowerCase().startsWith("/account/"))
+    presenceData.details = "Setting up account...";
+  else if (document.location.pathname.startsWith("/gold"))
+    presenceData.details = "Viewing the Gold plan.";
+  else if (document.location.pathname.startsWith("/art"))
+    presenceData.details = "Viewing art creations.";
+  else if (document.location.pathname.startsWith("/cosplay"))
+    presenceData.details = "Viewing cosplay creations.";
+  else if (document.location.pathname.startsWith("/commissionsopen"))
+    presenceData.details = "Viewing open commissions.";
+  else if (
+    document.location.pathname.toLowerCase().startsWith("/home/featured")
+  )
+    presenceData.details = "Viewing featured creators.";
+  else if (document.location.pathname.toLowerCase() === "/explore")
+    presenceData.details = "Viewing the explore page.";
+  else if (document.location.pathname.toLowerCase().startsWith("/blog/"))
+    presenceData.details = "Creating a blog post...";
+  else if (document.location.pathname.toLowerCase().startsWith("/manage"))
+    presenceData.details = "Managing Ko-Fi";
+  else if (document.location.pathname.toLowerCase().startsWith("/settings"))
+    presenceData.details = "Adjusting user settings...";
+  else if (document.location.pathname.toLowerCase().startsWith("/newsfeed"))
+    presenceData.details = "Viewing the newsfeed...";
+  else if (
+    document.location.pathname.toLowerCase().startsWith("/my-supporters")
+  )
+    presenceData.details = "Viewing supporters...";
+  else if (document.location.pathname.toLowerCase().startsWith("/streamalerts"))
+    presenceData.details = "Viewing stream alerts.";
+  else if (document.location.pathname.toLowerCase().startsWith("/shop"))
+    presenceData.details = "Viewing shop.";
+  else if (document.location.pathname.toLowerCase() === "/about")
+    presenceData.details = "Viewing Ko-Fi's About Page";
+  else if (document.location.pathname.toLowerCase().startsWith("/s/")) {
+    try {
+      const shopItem = document
+          .querySelector(
+            "#shop-item-detail > div > div.kfds-lyt-between-algn-top-row-to-col.kfds-c-sticky > div.sidebar.kfds-c-sticky-wrapper.kfds-c-order-2.kfds-c-shop-detail-wrapper > div.kfds-lyt-width-100.kfds-c-lyt-pdg-16-24.kfds-c-shop-detail-column-control > span"
+          )
+          .textContent.split("Jump to")[0]
+          .trim(),
+        shopOwner = document
+          .querySelector(
+            "#shop-item-detail > div > div.kfds-lyt-between-algn-top-row-to-col.kfds-c-sticky > div.sidebar.kfds-c-sticky-wrapper.kfds-c-order-2.kfds-c-shop-detail-wrapper > div.kfds-lyt-width-100.kfds-c-lyt-pdg-16-24.kfds-c-shop-detail-column-control > div > a > div > span:nth-child(1)"
+          )
+          .textContent.split("Jump to")[0]
+          .trim(),
+        URLSplit = document.location.pathname.split("/"),
```

Comment:
use array deconstructing

Reply:
I'm not exactly sure what you mean in this context. Could you please give me an example 🥺

Reply:
> use array deconstructing

I think it's fine

Reply:
> I think it's fine

Let me know if I can close this conversation.

Reply:
I mean with URLSplit because you only use [2]

Reply:
So I tried a bunch of things as you can see and nothing worked so if you could please give me some sort of example of what you mean @Bas950 that would be great.